src/main/kotlin/io/github/nomisrev/routes/articles.kt

```diff
@@ -41,6 +71,35 @@ data class Comment(
   val author: Profile
 )
 
+@Resource("/article")
+data class ArticleResource(val parent: RootResource = RootResource) {
+  @Resource("/feed") data class Feed(val parent: ArticleResource = ArticleResource())
+}
+
+fun Route.articleRoutes(
+  articleService: ArticleService,
+  jwtService: JwtService,
+) {
+
+  get<ArticleResource.Feed> {
+    jwtAuth(jwtService) { (_, userId) ->
+      either {
+          val (limit, offset) = receiveCatching<ArticleWrapper<UserFeed>>().bind().article
+          val articlesFeed =
+            articleService.getUserFeed(input = GetFeed(userId, limit, offset)).bind()
```

Comment:
We can now validate our input using the above defined validation, and rely on the resulting `GetFeed`.

```suggestion
        val getFeed = feed.validate(userId).bind()

        val articlesFeed =
          articleService.getUserFeed(input = getFeed).bind()
```